cmake/sysbuild/partition_manager.cmake

```diff
@@ -0,0 +1,702 @@
+
+# Load static configuration if found.
+# Try user defined file first, then file found in configuration directory,
+# finally file from board directory.
+
+macro(add_region)
+  set(oneValueArgs NAME SIZE BASE PLACEMENT DEVICE DEFAULT_DRIVER_KCONFIG DYNAMIC_PARTITION)
+  cmake_parse_arguments(REGION "" "${oneValueArgs}" "" ${ARGN})
+  list(APPEND regions ${REGION_NAME})
+  list(APPEND region_arguments "--${REGION_NAME}-size;${REGION_SIZE}")
+  list(APPEND region_arguments "--${REGION_NAME}-base-address;${REGION_BASE}")
+  list(APPEND region_arguments
+    "--${REGION_NAME}-placement-strategy;${REGION_PLACEMENT}")
+  if (REGION_DEVICE)
+    list(APPEND region_arguments "--${REGION_NAME}-device;${REGION_DEVICE}")
+  list(APPEND region_arguments
+       "--${REGION_NAME}-default-driver-kconfig;${REGION_DEFAULT_DRIVER_KCONFIG}")
+  endif()
+  if (REGION_DYNAMIC_PARTITION)
+    list(APPEND region_arguments
+      "--${REGION_NAME}-dynamic-partition;${REGION_DYNAMIC_PARTITION}")
+  endif()
+endmacro()
+
+set(user_def_pm_static ${PM_STATIC_YML_FILE})
+
+ncs_file(CONF_FILES ${APPLICATION_CONFIG_DIR}
+         PM conf_dir_pm_static
+         DOMAIN ${DOMAIN}
+         BUILD ${CONF_FILE_BUILD_TYPE}
+)
+
+ncs_file(CONF_FILES ${BOARD_DIR}
+         PM board_dir_pm_static
+         DOMAIN ${DOMAIN}
+         BUILD ${CONF_FILE_BUILD_TYPE}
+)
+
+if(EXISTS "${user_def_pm_static}" AND NOT IS_DIRECTORY "${user_def_pm_static}")
+  set(static_configuration_file ${user_def_pm_static})
+elseif (EXISTS ${conf_dir_pm_static})
+  set(static_configuration_file ${conf_dir_pm_static})
+elseif (EXISTS ${board_dir_pm_static})
+  set(static_configuration_file ${board_dir_pm_static})
+endif()
+
+if (EXISTS ${static_configuration_file})
+  message(STATUS "Found partition manager static configuration: "
+                 "${static_configuration_file}"
+  )
+  set(static_configuration --static-config ${static_configuration_file})
+endif()
+
+if (NOT static_configuration AND CONFIG_PM_IMAGE_NOT_BUILT_FROM_SOURCE)
+  message(WARNING
+    "One or more child image is not configured to be built from source. \
+    However, there is no static configuration provided to the \
+    partition manager. Please provide a static configuration as described in \
+    the 'Scripts -> Partition Manager -> Static configuration' chapter in the \
+    documentation. Without this information, the build system is not able to \
+    place the image correctly in flash.")
```

Comment:
Don't think it is possible to have these conditions in sysbuild? The option to not build from source does not exist

Reply:
not atm, but I think the possibility of providing a pre-built image when using sysbuild is a valid use-case.

Question is what makes most sense in this regard.
- Keep the pm code for sysbuild integration as close to the original code, even when certain use-cases are not yet supported.
- Remove code for currently unsupported use-cases, just to re-introduce the code when we start supporting it.
   This strategy implies a little higher risk of forgetting to re-introduce this code compared to have it as dormant code.

I can remove code not yet in use, but considering the above, do you have a preference ?